### PR TITLE
feat: Add ability to return json objects of all endpoints

### PR DESCRIPTION
**Pull request by sc0ttes**
Should be all good to go this time!

### CODE REVIEW

**Comment by smutel** [comment]
Thank you for this new PR.
Only one thing, could you please prefix your commit comment by `feat: `.

**Comment by smutel** [comment]
Hello, now your commit is correct but you have a merge item from master, I don't know why.

Could you please sync the master in your repo with the master in my repo (Add origin / git fetch origin / git merge origin/master) ?

Then go to your branch and type git rebase origin/master.

Thanks again.

**Comment by sc0ttes** [comment]
Hey Samuel,

Should be all fixed now. My master had been out of sync with yours so last time I got mine back in sync, I merged it into json_endpoints without rebasing afterwards.

**Comment by smutel** [comment]
Sorry you have some minor lint issues to resolve: https://github.com/smutel/terraform-provider-netbox/pull/19/checks#step:4:26

File: utils/netbox_generate_json_datasources.sh [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+package netbox
+
+import (
+        "encoding/json"
+
+        "github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+        netboxclient "github.com/netbox-community/go-netbox/netbox/client"
+)
+
+func dataNetboxJson${SECTION}${ITEM}List() *schema.Resource {
```
Comment (smutel) [inline utils/netbox_generate_json_datasources.sh:22]:
```suggestion
func dataNetboxJSON${SECTION}${ITEM}List() *schema.Resource {
```

File: utils/netbox_generate_json_datasources.sh [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+        return &schema.Resource{
+                Read: dataNetboxJson${SECTION}${ITEM}ListRead,
```
Comment (smutel) [inline utils/netbox_generate_json_datasources.sh:24]:
```suggestion
                Read: dataNetboxJSON${SECTION}${ITEM}ListRead,
```

File: utils/netbox_generate_json_datasources.sh [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+                Schema: map[string]*schema.Schema{
+                        "json": {
+                                Type:     schema.TypeString,
+                                Computed: true,
+                        },
+                },
+        }
+}
+
+func dataNetboxJson${SECTION}${ITEM}ListRead(d *schema.ResourceData, m interface{}) error {
```
Comment (smutel) [inline utils/netbox_generate_json_datasources.sh:35]:
```suggestion
func dataNetboxJSON${SECTION}${ITEM}ListRead(d *schema.ResourceData, m interface{}) error {
```

File: utils/netbox_generate_json_datasources.sh [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+
+        list, err := client.${SECTION}.${SECTION}${ITEM}List(nil, nil)
+        if err != nil {
+                return err
+        }
+
+        j, _ := json.Marshal(list.Payload.Results)
+
+        d.Set("json", string(j))
+        d.SetId("NetboxJson${SECTION}${ITEM}List")
```
Comment (smutel) [inline utils/netbox_generate_json_datasources.sh:46]:
```suggestion
        d.SetId("NetboxJSON${SECTION}${ITEM}List")
```

File: utils/netbox_generate_json_datasources.sh [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+
+        return nil
+}
+EOF
+
+
+cat << EOF >> provider_update.txt
+"netbox_json_${ENDPOINT}_list": dataNetboxJson${SECTION}${ITEM}List(),
```
Comment (smutel) [inline utils/netbox_generate_json_datasources.sh:54]:
```suggestion
"netbox_json_${ENDPOINT}_list": dataNetboxJSON${SECTION}${ITEM}List(),
```

**Review by smutel — CHANGES_REQUESTED** [review]

**Comment by smutel** [comment]
@sc0ttes  ?

**Comment by smutel** [comment]
Thank your contribution. I rewrote some part in #37.

**Comment by sc0ttes** [comment]
Sorry got really busy at work but thanks for merging!!

**Pull request closed without merging**